core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MergeCustomHeadersStage.java

```diff
@@ -52,7 +54,15 @@ public MergeCustomHeadersStage(HttpClientDependencies dependencies) {
     private final Map<String, List<String>> mergeHeaders(Map<String, List<String>>... headers) {
         Map<String, List<String>> result = new LinkedHashMap<>();
         for (Map<String, List<String>> header : headers) {
-            header.forEach((k, v) -> result.computeIfAbsent(k, ignored -> new ArrayList<>()).addAll(v));
+            header.forEach((headerName, headerValues) -> {
+                List<String> resultHeaderValues = result.computeIfAbsent(headerName, ignored -> new ArrayList<>());
+                if (SdkHttpUtils.isSingleHeader(headerName)) {
+                    resultHeaderValues.clear();
+                    resultHeaderValues.add(CollectionUtils.lastElement(headerValues));
```

Comment:
Should we disallow (or log) when there are multiple values specified in the same headerValues for a single-header?

It's a bit weird if I add multiple content-types at the request level, but it just takes the last one.

Reply:
I went back and forth with this. Alternatively what do you think about keeping all the values in the current list, but not merging them with the previously seen list?

e.g.

"FooHeader" set by marshaller -> "a"
"FooHeader" override on client -> ["b", "c"]
"FooHeader" override on request -> ["d", "e", "f"]

"FooHeader" on the wire -> ["d", "e", "f"]

Reply:
After discussing offline, went with the above solution.